celery/result.py

```diff
@@ -993,11 +993,7 @@ def __copy__(self):
     def ready(self):
         return True
 
-    def get(self, timeout=None, propagate=True,
-            disable_sync_subtasks=True, **kwargs):
```

Comment:
this change seems to be backward incompatible

Reply:
Do you mean some users would expect the assertion, or use `disable_sync_subtasks` as a positional argument? Otherwise, as a keyword argument it's consumed in `kwargs`.

I could leave the parameter and remove the conditional assertion.

Reply:
@auvipy I pushed a version that leaves the parameter list unchanged, now fully backwards compatible.

I don't understand @thedrow 's response. `EagerResult` is about a result where the task is already executed and finished, disabling sync subtasks makes no sense to me once you already have the result.

Reply:
But what if that eager result depends on `AsyncResult`s?

Reply:
`EagerResult.get` is so short that I can quote it in full:

```python
def get(self, propagate=True, disable_sync_subtasks=True):
    # << there is currently a check here about disable_sync_subtasks that I'd like to remove
    if self.successful():  
        return self.result  
    elif self.state in states.PROPAGATE_STATES:  
        if propagate:  
            raise self.result if isinstance(  
                self.result, Exception) else Exception(self.result)  
        return self.result
```

This accesses the following attributes of `self` :
 - `def successful(self): return self.state == states.SUCCESS`
 - `EagerResult.state`: a property, returns `self._state`
 - `self.result`: returns `self._result`

None of these depend on other tasks. Therefore, `EagerResult.get` always returns a value or raises an exception in a handful of instructions, and there's no need to disable sync subtasks _inside it_. I'm _not_ talking about `task.apply`.

@thedrow Your response is appreciated, our application uses a not-nice workaround now.

Reply:
The problem arises from mixing sync and async tasks which is possible and a slight change in the could make an EagerResult an AsyncResult.
Since they have the same interface it's not easy to tell until you run the code. This PR can create confusing behavior.
The way to wait for a task to complete and execute another task is using a `chain`.